contentcuration/contentcuration/frontend/shared/views/channel/ChannelModal.vue

```diff
@@ -228,14 +251,59 @@
       // For some reason the 'hideScroll' method of the VDialog is not
       // being called the first time the dialog is opened, so do that explicitly
       this.$refs.dialog.hideScroll();
+
+      // Set expiry to 1ms
+      this.header = this.channel.name; // Get channel name when user enters modal
+      this.tracker = new ChangeTracker(1);
+      this.tracker.start();
     },
     methods: {
-      ...mapActions('channel', ['updateChannel', 'loadChannel']),
+      ...mapActions('channel', ['updateChannel', 'loadChannel', 'deleteChannel']),
       hideHTMLScroll(hidden) {
         document.querySelector('html').style = hidden
           ? 'overflow-y: hidden !important;'
           : 'overflow-y: auto !important';
       },
+      saveChannel() {
+        if (this.$refs.detailsform.validate()) {
+          this.tracker.stop();
+          this.tracker.dismiss();
+          this.changed = false;
+
+          if (this.channel.new) {
+            // Make sure channel gets created before navigating to channel
+            this.updateChannel({ id: this.channelId, new: false });
+            setTimeout(() => {
+              window.location = window.Urls.channel(this.channelId);
+            }, 6500);
```

Comment:
So this is based off the sync needing to happen first? Would it make sense to create a special view that checks every couple seconds if the channel is created, with a loading animation and message, then redirects?

Reply:
Ah sorry, that didn't end up working either. Forgot to remove it

Reply:
Updated!